### PR TITLE
Better multithreading performance

### PR DESCRIPTION
**Pull request by shagren**
if you read captures in threads app performance will be decreased due GIL locking.
This PR release GIL inside C functions if you use `thread_safe=False` mode.
In this case you can safety work with camera only from one thread.

### CODE REVIEW

File: pyk4a/pyk4a.py [hunk 1]
```diff
@@ -24,9 +24,10 @@ class K4ATimeoutException(K4AException):
 class PyK4A:
     TIMEOUT_WAIT_INFINITE = -1
 
-    def __init__(self, config=Config(), device_id=0):
+    def __init__(self, config=Config(), device_id=0, thread_safe: bool = True):
```
Comment (shagren) [inline pyk4a/pyk4a.py:27]: By default thread_safe behavior will be preserved

**Comment by lpasselin** [comment]
Very interesting, thanks for the PR!

@shagren Did you confirm it works with a kinect?

@etiennedub do you have time to test this?

**Comment by lpasselin** [comment]
Also did you notice significative improvements by releasing the gil?

**Comment by shagren** [comment]
@lpasselin , it works for my setup: linux and windows

I wrote some example for comparing performance, please check result:
image

**Comment by lpasselin** [comment]
Awesome PR, thanks a lot for this enhancement!

**Pull request merged**